From b7e7080b5c847ed3381bccb59ba95e8ca1a6c5f4 Mon Sep 17 00:00:00 2001
From: Mao Minkai <maominkai@wxiat.com>
Date: Tue, 6 Sep 2022 16:39:04 +0800
Subject: [PATCH] sw64: remove useless local r26 in setup_rt_frame()
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: b7e7080b5c847ed3381bccb59ba95e8ca1a6c5f4
Modified-by-SEL: No


Sunway inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I5XTGY

--------------------------------

Signed-off-by: Mao Minkai <maominkai@wxiat.com>
Reviewed-by: He Sheng <hesheng@wxiat.com>
Signed-off-by: Gu Zitao <guzitao@wxiat.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/sw_64/kernel/signal.c | 9 ++-------
 1 file changed, 2 insertions(+), 7 deletions(-)

diff --git a/arch/sw_64/kernel/signal.c b/arch/sw_64/kernel/signal.c
index 32c9484d2aa2..b80cf0e56224 100644
--- a/arch/sw_64/kernel/signal.c
+++ b/arch/sw_64/kernel/signal.c
@@ -261,7 +261,7 @@ setup_sigcontext(struct sigcontext __user *sc, struct pt_regs *regs,
 static int
 setup_rt_frame(struct ksignal *ksig, sigset_t *set, struct pt_regs *regs)
 {
-	unsigned long oldsp, r26, err = 0;
+	unsigned long oldsp, err = 0;
 	struct rt_sigframe __user *frame;
 
 	oldsp = rdusp();
@@ -283,13 +283,8 @@ setup_rt_frame(struct ksignal *ksig, sigset_t *set, struct pt_regs *regs)
 	if (err)
 		return -EFAULT;
 
-	/* Set up to return from userspace.  If provided, use a stub
-	 * already in userspace.
-	 */
-	r26 = VDSO_SYMBOL(current->mm->context.vdso, rt_sigreturn);
-
 	/* "Return" to the handler */
-	regs->r26 = r26;
+	regs->r26 = VDSO_SYMBOL(current->mm->context.vdso, rt_sigreturn);
 	regs->r27 = regs->pc = (unsigned long) ksig->ka.sa.sa_handler;
 	regs->r16 = ksig->sig;                    /* a0: signal number */
 	if (ksig->ka.sa.sa_flags & SA_SIGINFO) {
-- 
2.33.0

